### DISCUSSION TITLE
Morph Relation

### DISCUSSION THREAD

**Discussion by soluzione-software** [Laravel-Backpack/community-forum#662] [Q&A (Help)]
Good morning,
i have an error in polimorphic relation:
I have 3 models:

class Producer extends Model
{
    public function user(): MorphOne
    {
        return $this->morphOne(User::class, 'userable');
    }
******
class Judge extends Model
{
    public function user(): MorphOne
    {
        return $this->morphOne(User::class, 'userable');
    }
*******
class User extends Authenticatable
{
    public function userable(): MorphTo{
        return $this->MorphTo(__FUNCTION__, 'userable_type', 'userable_id');
    }
****
my migration:
        Schema::table('users', function (Blueprint $table) {
            $table->morphs('userable');
        });
*****
when i call:
\App\Models\User::find(2)->userable 
if user havent any model related (userable_id and userable_type is empty) i get this error:
 SQLSTATE[42S22]: Column not found: 1054 Unknown column 'users.' in 'where clause' (SQL: select * from `users` where `users`.`` = 0 and `users`.`deleted_at` is null limit 1).

why?
can someone help me?
tanks

**Comment by karandatwani92** [comment]
Hey @soluzione-software 

Sorry, This forum is for backpack-related queries only.
I suggest you post your query on [StackOverflow](https://stackoverflow.com/).

Check if the following resource helps.
- https://laraveldaily.com/post/laravel-belongstomany-polymorphic-relations-practical-example
- https://stackoverflow.com/questions/27866020/laravel-returning-the-namespaced-owner-of-a-polymorphic-relation/27909753#27909753
